test/test_smoke.py

```diff
@@ -0,0 +1,67 @@
+import json
+import os
+import pathlib
+import subprocess
+
+import pytest
+
+# local test utils
+import testutil
+
+
+@pytest.fixture(name="output_path")
+def output_path_fixture(tmp_path):
+    output_path = tmp_path / "output"
+    output_path.mkdir(exist_ok=True)
+    return output_path
+
+
+@pytest.fixture(name="config_json")
+def config_json_fixture(output_path):
+    CFG = {
+        "blueprint": {
+            "customizations": {
+                "user": [
+                    {
+                        "name": "test",
+                        "password": "password",
+                        "groups": ["wheel"],
+                    },
+                ],
+            },
+        },
+    }
+    config_json_path = output_path / "config.json"
+    config_json_path.write_text(json.dumps(CFG), encoding="utf-8")
+    return config_json_path
+
+
+@pytest.mark.skipif(os.getuid() != 0, reason="needs root")
+@pytest.mark.skipif(not testutil.has_executable("podman"), reason="need podman")
+def test_smoke(output_path, config_json):
+    # build local container
```

Comment:
The interesting thing here of course is that I think it'd also make sense to support a flow that doesn't provision an intermediate VM and rebuild the container image in it, but just executes on an already built container. 

Among other things, that's the clear pattern we want to use for things like reverse dependency testing.

I think tmt in theory has some support for this.

Reply:
@ondrejbudai would be great to capture these expected use cases, probably in the Feature, so we know what to test against - when you are back.